diff --git a/red.js b/red.js
index 6162add..4a85030 100755
--- a/red.js
+++ b/red.js
@@ -25,6 +25,7 @@ var nopt = require("nopt");
 var path = require("path");
 var fs = require("fs-extra");
 var RED = require("./red/red.js");
+require('systemd');
 
 var server;
 var app = express();
@@ -254,7 +255,8 @@ RED.start().then(function() {
             }
             process.exit(1);
         });
-        server.listen(settings.uiPort,settings.uiHost,function() {
+        server.listen(settings.uiPort);
+        server.on('listening', function() {
             if (settings.httpAdminRoot === false) {
                 RED.log.info(RED.log._("server.admin-ui-disabled"));
             }
diff --git a/settings.js b/settings.js
index a6e6797..4437acd 100644
--- a/settings.js
+++ b/settings.js
@@ -18,9 +18,11 @@
 // to make it available:
 //var fs = require("fs");
 
+var port = process.env.LISTEN_PID > 0 ? 'systemd' : 1880;
+
 module.exports = {
     // the tcp port that the Node-RED web server is listening on
-    uiPort: 1880,
+    uiPort: port,
 
     // By default, the Node-RED UI accepts connections on all IPv4 interfaces.
     // The following property can be used to listen on a specific interface. For
